hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferUtils.java

```diff
@@ -468,38 +468,55 @@ public static void writeVLong(ByteBuffer out, long i) {
     }
   }
 
-  private interface ByteVisitor {
-    byte get();
-  }
-
-  private static long readVLong(ByteVisitor visitor) {
-    byte firstByte = visitor.get();
+  /**
```

Comment:
Nit: could we keep using the Visitor pattern here? It seems both methods only differ in two lines.

Reply:
I guess removing the visitor could also increase the performance a bit, maybe...

Reply:
When I was iterating on this, I did find that splitting out from the visitor increased performance a bit. I didn't save the intermediate JMH benchmark results/code (I can re-run the tests if you'd like, but I can certainly generate those numbers again.